src/api/procedures/__tests__/issueTokens.ts

```diff
@@ -131,53 +129,48 @@ describe('setIssuancesData procedure', () => {
     const proc = procedureMockUtils.getInstance<Params, SecurityToken>();
     proc.context = mockContext;
 
-    return expect(prepareSetIssuancesData.call(proc, args)).rejects.toThrow(
-      `The total supply for "${ticker}" cannot be bigger than ${limitTotalSupply.toString()}`
+    return expect(prepareIssueTokens.call(proc, args)).rejects.toThrow(
+      `This issuance operation will cause the total supply of "${ticker}" to exceed the maximum allowed (${MAX_TOKEN_AMOUNT.toFormat()})`
     );
   });
 
   test('should add a batch issue transaction to the queue', async () => {
-    const balanceOne = 100;
-    const balanceTwo = 200;
     const args = {
-      issuances: [
+      issuanceData: [
         {
           did: 'someDid',
-          balance: new BigNumber(balanceOne),
+          amount: new BigNumber(100),
         },
         {
           did: 'otherDid',
-          balance: new BigNumber(balanceTwo),
+          amount: new BigNumber(200),
         },
       ],
       ticker,
     };
 
-    const someIdentityId = 'someIdentityId';
-    const otherIdentityId = 'otherIdentityId';
-
-    const investors = [
-      polkadotMockUtils.createMockIdentityId(someIdentityId),
-      polkadotMockUtils.createMockIdentityId(otherIdentityId),
-    ];
-    const balances = [
-      polkadotMockUtils.createMockBalance(balanceOne),
-      polkadotMockUtils.createMockBalance(balanceTwo),
-    ];
+    const investors: IdentityId[] = [];
+    const balances: Balance[] = [];
 
     const stringToIdentityIdStub = sinon.stub(utilsModule, 'stringToIdentityId');
     const numberToBalanceStub = sinon.stub(utilsModule, 'numberToBalance');
 
-    stringToIdentityIdStub.withArgs(args.issuances[0].did, mockContext).returns(investors[0]);
-    stringToIdentityIdStub.withArgs(args.issuances[1].did, mockContext).returns(investors[1]);
-    numberToBalanceStub.withArgs(args.issuances[0].balance, mockContext).returns(balances[0]);
-    numberToBalanceStub.withArgs(args.issuances[1].balance, mockContext).returns(balances[1]);
+    args.issuanceData.forEach(data => {
```

Comment:
You should destructure data here instead of doing `data.did` and `data.amount` twice